src/commands/cmd_delete.c

```diff
@@ -8,93 +8,86 @@
 #include "cmd_delete.h"
 
 #include <assert.h>
-#include "stores/store.h"
 #include "../graph/graph.h"
 #include "../query_executor.h"
 #include "../util/simple_timer.h"
 
-/* DeleteGraphContext is used to hold all information required 
- * in order to delete a graph. */
+extern RedisModuleType *GraphContextRedisModuleType;
+
+/* DeleteContext contains the Redis string used as the graph's key
+ * and a blocked Redis client to interact with. */
 typedef struct {
-    RedisModuleString *graphID;     /* Graph ID been deleted. */
+    RedisModuleString *graph_name;  /* Name of graph to delete. */
     RedisModuleBlockedClient *bc;   /* Redis blocked client. */
-} DeleteGraphContext;
+} DeleteContext;
 
-DeleteGraphContext *DeleteGraphContext_new(RedisModuleString *graphID, RedisModuleBlockedClient *bc) {
-    DeleteGraphContext *ctx = malloc(sizeof(DeleteGraphContext));
+DeleteContext* _DeleteContext_New(RedisModuleString *graph_name, RedisModuleBlockedClient *bc) {
+    DeleteContext *ctx = malloc(sizeof(DeleteContext));
     ctx->bc = bc;
-    ctx->graphID = graphID;
+    ctx->graph_name = graph_name;
     return ctx;
 }
 
-void DeleteGraphContext_free(DeleteGraphContext *ctx) {
+void _DeleteContext_Free(DeleteContext *ctx) {
     free(ctx);
 }
 
-/* Delete graph, removes every Redis key related to given graph 
- * free every resource allocated by the graph. */
+/* Delete graph, removing the key from Redis and
+ * freeing every resource allocated by the graph. */
 void _MGraph_Delete(void *args) {
```

Comment:
This function is a fair bit better now. It replies with appropriate errors, won't erase non-Graph keys if given a name associated with a different key, etc. I think the concurrent delete tests consistently succeed now, too!